kernel/kernel-api/src/test/scala/io/delta/kernel/internal/replay/ActionsIteratorSuite.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.internal.replay
+
+import java.util.{Collections, Optional}
+import java.util.NoSuchElementException
+
+import io.delta.kernel.engine.FileReadResult
+import io.delta.kernel.expressions.Predicate
+import io.delta.kernel.test.{BaseMockParquetHandler, MockEngineUtils}
+import io.delta.kernel.types.StructType
+import io.delta.kernel.utils.{CloseableIterator, FileStatus}
+
+import org.scalatest.funsuite.AnyFunSuite
+
+class ActionsIteratorSuite extends AnyFunSuite with MockEngineUtils {
```

Comment:
actually, I think in the issue it said it was quite easy to repro this with spark? can we add a test in `kernelDefaults` that does this with spark?